proposals/vector_to_regular_points.json

```diff
@@ -0,0 +1,41 @@
+{
+    "id": "vector_to_regular_points",
+    "summary": "Sample regular points from geometries",
+    "description": "Generate a vector data cube of points by sampling regularly-spaced points from input geometries.",
+    "categories": [
+        "cubes",
+        "vector"
+    ],
+    "experimental": true,
+    "parameters": [
+        {
+            "name": "data",
+            "description": "Input geometries for sample extraction. Vector properties are preserved for vector data cubes and all GeoJSON Features.\n\nTo maximize interoperability, a nested `GeometryCollection` should be avoided. Furthermore, a `GeometryCollection` composed of a single type of geometries should be avoided in favour of the corresponding multi-part type (e.g. `MultiPolygon`).",
+            "schema": [
+                {
+                    "type": "object",
+                    "subtype": "geojson"
+                },
+                {
+                    "type": "object",
+                    "subtype": "vector-cube"
+                }
+            ]
+        },
+        {
+            "name": "distance",
+            "description": "Defines the minimum distance in degrees that is required between two samples generated *inside* a single geometry.\n\n- For **polygons**, the distance defines the cell sizes of a regular grid that starts at the upper-left bound of each polygon. The centroid of each cell is then a sample point. If the centroid is not enclosed in the polygon, no point is sampled. If no point can be sampled for the geometry at all, the first coordinate of the geometry is returned as point.\n- For **lines** (line strings), the sampling starts with a point at the first coordinate of the line and then walks along the line and samples a new point each time the distance to the previous point has been reached again.\n- For **points**, the point is returned as given.",
```

Comment:
> minimum distance in degrees

What kind of degrees is that? longitude/latitude? E.g. at the latitude of Belgium, a latitude degree is almost 2 times the distance on the ground of a longitude degree.

> the cell sizes of a regular grid

I guess a rectangular/square grid is meant (and not a triangular/hexagonal). I think it is important to mention that.

> regular grid that starts at the upper-left bound of each polygon.

Is there a particular reason to state that? I think it is computationally more efficient, predictable and stable to use the same grid for all geometries (e.g. aligned with the coordinate ref system)

Reply:
1) Goes back to the question about units, we need to decide on a common way here. Pops up now in mostly every vector process PR, will open a separate issue.

2) Yes, will clarify.

3) Is it? I guess, we could replace "each" with polygon "all polygons combined". Originated here, I think https://github.com/Open-EO/openeo-processes/pull/315#discussion_r813034467

Reply:
> What kind of degrees is that? longitude/latitude? E.g. at the latitude of Belgium, a latitude degree is almost 2 times the distance on the ground of a longitude degree.

Yes, but since we are limited by geoJSON which supports WGS84 only that's the only way to define it at the moment. When we'll have vector-cubes with different CRSs we could use meters instead. Should we consider this already?

> Is there a particular reason to state that? I think it is computationally more efficient, predictable and stable to use the same grid for all geometries (e.g. aligned with the coordinate ref system)

Good idea. Basically we would need to compute a single regular grid for the whole area of interest and then check which points fall in which geometry.

Reply:
We are not limited by GeoJSON. The schema includes vector-cubes. We should consider more, yes.